assets/src/components/customers/NewCustomerModal.tsx

```diff
@@ -0,0 +1,126 @@
+import React from 'react';
+import {Box} from 'theme-ui';
+import {Button, Input, Modal, Text} from '../common';
+import * as API from '../../api';
+import logger from '../../logger';
+
+const NewCustomerModal = ({
+  visible,
+  onSuccess,
+  onCancel,
+}: {
+  visible: boolean;
+  onSuccess: (params: any) => void;
+  onCancel: () => void;
+}) => {
+  const [accountId, setAccountId] = React.useState('');
+  const [email, setEmailAddress] = React.useState('');
+  const [name, setName] = React.useState('');
+  const [phone, setPhoneNumber] = React.useState('');
+  const [error, setErrorMessage] = React.useState<string | null>(null);
+  const [isSaving, setIsSaving] = React.useState(false);
+
+  React.useEffect(() => {
+    API.fetchAccountInfo()
+      .then(({id: accountId}) => setAccountId(accountId))
+      .catch((err) => logger.error('Failed to fetch account:', err));
+  }, []);
+
+  const handleChangeEmail = (e: React.ChangeEvent<HTMLInputElement>) =>
+    setEmailAddress(e.target.value);
+  const handleChangeName = (e: React.ChangeEvent<HTMLInputElement>) =>
+    setName(e.target.value);
+  const handleChangePhone = (e: React.ChangeEvent<HTMLInputElement>) =>
+    setPhoneNumber(e.target.value);
+
+  const resetInputFields = () => {
+    setName('');
+    setEmailAddress('');
+    setPhoneNumber('');
+    setErrorMessage(null);
+  };
+
+  const handleCancelCustomer = () => {
+    onCancel();
+    resetInputFields();
+  };
+
+  const handleCreateCustomer = async () => {
+    setIsSaving(true);
+
+    return API.createNewCustomer(accountId, {name, email, phone})
+      .then((result) => {
+        onSuccess(result);
+        resetInputFields();
+      })
+      .catch((err) => {
+        logger.error('Error creating customer:', err);
+        // TODO
+        const errorMessage = 'Invalid customer details!';
+        setErrorMessage(errorMessage);
+      })
+      .finally(() => setIsSaving(false));
+  };
```

Comment:
Is there a reason why we're handling the promise traditionally with then/catch/finally vs awaiting the promise and wrapping the logic in a try/catch?

Reply:
no reason really, i find both approaches interchangeable in many cases 🤷